presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -998,7 +997,10 @@ private Object nullInputEvaluator(final Collection<Symbol> nullSymbols, Expressi
                     emptyList(), /* parameters have already been replaced */
                     WarningCollector.NOOP);
             return ExpressionInterpreter.expressionOptimizer(expression, metadata, session, expressionTypes)
-                    .optimize(symbol -> nullSymbols.contains(symbol) ? null : symbol.toSymbolReference());
+                    .optimize(symbol -> nullVariables.stream()
```

Comment:
Do we want to move the conversion from `Collection<VariableReferenceExpression>` to outside the lambda? We only need to do the conversion once then.